src/main/java/com/hubspot/ringleader/watcher/PersistentWatcher.java

```diff
@@ -168,98 +174,16 @@ public Void apply(EventListener listener) {
     });
   }
 
-  private synchronized void replaceCurator() {
-    long timestamp = curatorTimestamp.get();
-    long age = System.currentTimeMillis() - timestamp;
-    long minAge = TimeUnit.MINUTES.toMillis(1);
-
-    // only attempt reconnect once per minute so we don't exacerbate a failure scenario and don't reconnect when the
-    // executor is shutting down.
-    if (age < minAge || !curatorTimestamp.compareAndSet(timestamp, System.currentTimeMillis()) || executor.isShutdown()) {
-      return;
-    }
-
-    executor.submit(new Runnable() {
-
-      //@Override Java 5 compatibility
+  private void replaceCurator(final boolean backgroundFetch) {
+    parent.replaceCurator(new Runnable() {
+      @Override
       public void run() {
-        CuratorFramework previous = curatorReference.getAndSet(newCurator());
-        cleanup(previous);
-        fetch(false);
-      }
-    });
-  }
-
-  private CuratorFramework newCurator() {
-    CuratorFramework curator = null;
-    try {
-      curator = curatorSupplier.get();
-      if (curator.getState() != CuratorFrameworkState.STARTED) {
-        curator.start();
-      }
-
-      curator.getConnectionStateListenable().addListener(new ConnectionStateListener() {
-
-        //@Override Java 5 compatibility
-        public void stateChanged(CuratorFramework client, ConnectionState newState) {
-          switch (newState) {
-            case SUSPENDED:
-            case LOST:
-              LOG.error("Connection lost or suspended, replacing client");
-              replaceCurator();
-              break;
-            default:
-              // make findbugs happy
+        executor.submit(new Runnable() {
+          @Override
+          public void run() {
+            fetch(backgroundFetch);
```

Comment:
Out of curiosity, why resubmit to the executor vs. just running now?

Reply:
the real reason is because that's what the old code did, but we could probably just call it synchronously